src/dateparser/dateparser.js

```diff
@@ -124,8 +124,10 @@ angular.module('ui.bootstrap.dateparser', [])
         results = input.match(regex);
 
     if ( results && results.length ) {
-      var fields, dt;
-      if (baseDate) {
+      var fields, dt,
+          isValidBaseDate = baseDate && !isNaN(baseDate.valueOf());
```

Comment:
This doesn't look quite right - this should probably be `angular.isDate(baseDate) && !isNaN(baseDate.getTime())`.